Bz909 possible atom leak

kellymclaughlin opened the pull request:
This fixes the potential atom leak described in BZ909 and also adds some improved checking and error feedback for invalid RW, W, and DW values.

kellymclaughlin committed:
Use binary_to_existing_atom in riak_kv_util:normalize_rw_value/2 to
prevent a potential atom leak. Also add appropriate exception handling
around call to binary_to_existing_atom and add some extra test cases
for the unit test for normalize_rw_value.

kellymclaughlin committed:
Check for invalid values for W and DW.

kellymclaughlin committed:
Check for invalid RW value.

slfritchie commented:
Kelly: would you mind adding some eunit tests for the put & delete FSMs?

It's interesting that the get FSM doesn't require mods: atom() > integer()
is always true.

kellymclaughlin committed:
Reorganize the existing eunit test for put fsm and add more eunit
tests to test success and failure conditions when starting the fsm.

kellymclaughlin committed:
Add eunit test for riak_kv_delete:delete/6.

kellymclaughlin committed:
Add some delays during the setup for the put_fsm and delete unit tests
to ensure that all the applications have properly started and avoid
spurious test failures.

kellymclaughlin committed:
Clear riak_core environment variable vnode_modules during cleanup
to ensure that riak_core restarts cleanly during unit tests.

kellymclaughlin committed:
Explicitly use short names when making eunit test node a distributed node.

slfritchie commented:
I saw 10 clean "make eunit" runs in a row, so I'd say it's good to merge.

slfritchie commented:
Hrm, I don't know why GitHub's auto-magic didn't close this
pull request, but it looks like Kelly merged his branch to "master",
so I'll close this.
